localstack/services/sns/sns_listener.py

```diff
@@ -65,47 +65,48 @@ def forward_request(self, method, path, data, headers):
             elif req_action == 'Publish':
                 message = req_data['Message'][0]
                 sqs_client = aws_stack.connect_to_service('sqs')
-                for subscriber in SNS_SUBSCRIPTIONS[topic_arn]:
-                    filter_policy = json.loads(subscriber.get('FilterPolicy', '{}'))
-                    message_attributes = get_message_attributes(req_data)
-                    if check_filter_policy(filter_policy, message_attributes):
-                        if subscriber['Protocol'] == 'sqs':
-                            endpoint = subscriber['Endpoint']
-                            if 'sqs_queue_url' in subscriber:
-                                queue_url = subscriber.get('sqs_queue_url')
-                            elif '://' in endpoint:
-                                queue_url = endpoint
-                            else:
-                                queue_name = endpoint.split(':')[5]
-                                queue_url = aws_stack.get_sqs_queue_url(queue_name)
-                                subscriber['sqs_queue_url'] = queue_url
-                            try:
-                                sqs_client.send_message(
-                                    QueueUrl=queue_url,
-                                    MessageBody=create_sns_message_body(subscriber, req_data)
+                if topic_arn is not None:
+                    for subscriber in SNS_SUBSCRIPTIONS[topic_arn]:
```

Comment:
nitpick: Can we remove line 68 and replace line 69 with:
```
for subscriber in SNS_SUBSCRIPTIONS.get(topic_arn) or []:
```
This would (1) help reducing the level of nesting, and (2) avoid the general case of a `KeyError` if a `topic_arn` is specified that is not contained in `SNS_SUBSCRIPTIONS`.

Reply:
From my understanding, the following line will fail with KeyError if topic_arn is not in the SNS_SUBSCRIPTIONS.
```
for subscriber in SNS_SUBSCRIPTIONS.get(topic_arn) or []:
```
I change it by:
```
for subscriber in SNS_SUBSCRIPTIONS.get(topic_arn,[]):
```

Reply:
> From my understanding, the following line will fail with KeyError if topic_arn is not in the SNS_SUBSCRIPTIONS

Not quite, but your implementation is also fine. Thanks!